src/policy/compiler.rs

```diff
@@ -1048,6 +1056,7 @@ where
 /// Helper function to compile different types of binary fragments.
 /// `sat_prob` and `dissat_prob` represent the sat and dissat probabilities of
 /// root or. `weights` represent the odds for taking each sub branch
+#[allow(clippy::too_many_arguments)]
```

Comment:
In e491308b6075d5274917750ad7258cb604b99aea:

I think this is fine, but FYI both this and the "too many arguments" lint thresholds can be set globally in clippy.toml.